digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataexport/AssetReportCsvExporterSpec.scala

```diff
@@ -0,0 +1,160 @@
+import fi.liikennevirasto.digiroad2.asset.{Municipality, TrafficDirection}
+import fi.liikennevirasto.digiroad2.{CsvDataExporter, DigiroadEventBus, Point}
+import fi.liikennevirasto.digiroad2.client.vvh.{FeatureClass, VVHRoadlink}
+import fi.liikennevirasto.digiroad2.csvDataExporter.AssetReportCsvExporter
+import fi.liikennevirasto.digiroad2.dao.{MunicipalityDao, MunicipalityInfo, OracleUserProvider}
+import fi.liikennevirasto.digiroad2.dao.csvexporter.{AssetReport, AssetReporterDAO}
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.service.RoadLinkService
+import fi.liikennevirasto.digiroad2.user.{Configuration, User, UserProvider}
+import org.joda.time.DateTime
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{FunSuite, Matchers}
+import slick.driver.JdbcDriver.backend.Database.dynamicSession
+
+
+class AssetReportCsvExporterSpec extends FunSuite with Matchers {
+
+  lazy val properties: Properties = {
+    val props = new Properties()
+    props.load(getClass.getResourceAsStream("/digiroad2.properties"))
+    props
+  }
+
+  val csvSeparator = ";"
+  val newLine = "\r\n"
+
+  val mockEventBus = MockitoSugar.mock[DigiroadEventBus]
+  val mockRoadLinkService = MockitoSugar.mock[RoadLinkService]
+  val mockAssetReporterDAO = MockitoSugar.mock[AssetReporterDAO]
+  val mockMunicipalityDao = MockitoSugar.mock[MunicipalityDao]
+  val mockUserProvider = MockitoSugar.mock[OracleUserProvider]
+
+  val csvDataExporter = new CsvDataExporter( mockEventBus )
+
+  val assetReportCsvExporter = new AssetReportCsvExporter(mockRoadLinkService, mockEventBus, mockUserProvider){
+    override val assetReporterDAO: AssetReporterDAO = mockAssetReporterDAO
+    override val municipalityDao: MunicipalityDao = mockMunicipalityDao
+    override val userProvider: UserProvider = mockUserProvider
+  }
+
+  val LupaMunicipalityInfo = MunicipalityInfo(408, 3, "Lapua")
+  val SavitaipaleMunicipalityInfo = MunicipalityInfo(739, 8, "Savitaipale")
+
+  val vvhRoadlinks1 = Seq(VVHRoadlink(388562360L, 408, Seq(Point(0, 0), Point(10, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))
+  val vvhRoadlinks2 = Seq(VVHRoadlink(38856690L, 739, Seq(Point(10, 0), Point(65, 0)), Municipality, TrafficDirection.UnknownDirection, FeatureClass.AllOthers))
+
+  val operatorUser = User(26867454, "k647320", Configuration(None, None, None, None, None, Set(), Set(), Set("operator"), Some("2018-11-27"), None), None)
+
+  val users = List( operatorUser,
+                  User(642513, "l485693", Configuration(None, None, None, None, None, Set(408, 739), Set(), Set(), None, None), None),
+                  User(123456, "lx775599", Configuration(None, None, None, None, None, Set(), Set(), Set("elyMaintainer"), Some("2019-12-07"), None), None) )
+
+
+  test("test function exportAssetsByMunicipality") {
+    OracleDatabase.withDynTransaction {
```

Comment:
Instead of doing this and at the end using ``` dynamicSession.rollback()``` it looks better with
``` Scala
private def runWithRollback(test: => Unit): Unit = TestTransactions.runWithRollback()(test)
```
And use ```runWithRollback``` just like we do in the other tests

Reply:
Done.